### PR TITLE
Feat: Event View Backend

### PR DESCRIPTION
**Pull request by OdeChan**
Improved backend of the get method endpoint of "eventview" to show actual event details instead of placeholders.

On the way:
- Fixed up get_event_by_id (previously by_id), including error handling
- Fixed delete_event and update_event to fit new get_event_by_id function
- Moved logger to dependencies so can be used outside of main
- Fixed up the event_test_client
- Added and fixed up event tests

### CODE REVIEW

File: app/routers/event.py [hunk 1]
```diff
@@ -25,16 +27,64 @@ async def eventedit(request: Request):
...
+@router.get("/view/{event_id}")
+async def eventview(request: Request, event_id: int,
+                    db: Session = Depends(get_db)):
+    try:
+        event = get_event_by_id(db, event_id)
+    except NoResultFound:
+        raise HTTPException(status_code=404, detail="Event not found")
+    except MultipleResultsFound:
+        raise HTTPException(status_code=500, detail="Multiple events found")
+    start_format = '%A, %d/%m/%Y %H:%M'
```
Comment (sagizaidor) [inline app/routers/event.py:39]: למה יש גם את שם החודש וגם את מספר החודש?

File: app/routers/event.py [hunk 2]
```diff
@@ -113,38 +166,10 @@ def get_participants_emails_by_event(db: Session, event_id: int) -> List[str]:
     """Returns a list of all the email address of the event invited users,
         by event id."""
 
-    return [email[0] for email in db.query(User.email).
+    return (
+        [email[0] for email in db.query(User.email).
             select_from(Event).
```
Comment (sagizaidor) [inline app/routers/event.py:171]: תהיה: אולי שווה לפתוח על זה טבלה בDB?

**Review by sagizaidor — COMMENTED** [review]
נראה אחלה,
לא היו יותר מידי דברים שתפסו את תשומת ליבי

File: app/routers/event.py (hunk 1 above)
Comment (OdeChan) [reply app/routers/event.py:39]: @sagizaidor %A is the is the week day, %m is the month and %M is minutes. Am I missing something?

File: app/routers/event.py (hunk 2 above)
Comment (OdeChan) [reply app/routers/event.py:171]: @sagizaidor I'm not actually sure, this isn't my code, but lint check failed me for it so I had to fix

File: app/routers/event.py (hunk 1 above)
Comment (sagizaidor) [reply app/routers/event.py:39]: אוי...תכלס, טעות שלי, התבלבלתי עם %B,...

File: app/routers/event.py [hunk 3]
```diff
@@ -25,16 +27,66 @@ async def eventedit(request: Request):
...
+        event = get_event_by_id(db, event_id)
+    except NoResultFound:
+        raise HTTPException(status_code=404, detail="Event not found")
+    except MultipleResultsFound:
+        raise HTTPException(status_code=500, detail="Multiple events found")
+
+    participants = get_participants_emails_by_event(db, event_id)
+
+    db.delete(event)  # Delete event
+    db.query(UserEvent).filter_by(event_id=event_id) \
```
Comment (yammesicka) [inline app/routers/event.py:62]: Prefer not to use backslash for the split. In this case, you can merge the lines. (You can remove the comment)

File: app/routers/event.py [hunk 4]
```diff
@@ -25,16 +27,66 @@ async def eventedit(request: Request):
...
+    if not isinstance(db, Session):
+        raise AttributeError(
+            f'Could not connect to database. '
+            f'db instance type received: {type(db)}')
+    try:
+        event = db.query(Event).filter_by(id=event_id).one()
+    except NoResultFound:
+        raise NoResultFound(f"Event ID does not exist. ID: {event_id}")
+    except MultipleResultsFound:
+        error_message = f'Multiple results found when getting event. ' \
```
Comment (yammesicka) [inline app/routers/event.py:85]: Prefer parenthesis here

File: app/routers/event.py [hunk 5]
```diff
@@ -25,16 +27,66 @@ async def eventedit(request: Request):
                                       {"request": request})
 
 
-@router.get("/view/{id}")
-async def eventview(request: Request, id: int):
+@router.get("/view/{event_id}")
```
Comment (yammesicka) [inline app/routers/event.py:30]: Is `/view/` needed?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/templates/event/partials/view_event_details_tab.html [hunk 6]
```diff
@@ -1,32 +1,30 @@
...
 </div>
 
-<div class="event_info_row">
-    <span class="icon">ICON</span>
-    <span>Repeats every INTERVAL</span>
-</div>
+<!--<div class="event_info_row">-->
+<!--    <span class="icon">ICON</span>-->
+<!--    <span>Repeats every INTERVAL</span>-->
+<!--</div>-->
```
Comment (ron-huberfeld) [inline app/templates/event/partials/view_event_details_tab.html:21]: האם זה תורם להשאיר את ההערות בקוד?

Comment (OdeChan) [reply app/templates/event/partials/view_event_details_tab.html:21]: Yes, these are placeholders till there are matching model fields

File: app/routers/event.py (hunk 2 above)
Comment (efratush) [reply app/routers/event.py:174]: לכל משתמש יש אימייל ויש טבלה מקשרת בין האירוע למשתמשים. איזה טבלה אתה מציע להוסיף?

File: app/routers/event.py (hunk 5 above)
Comment (OdeChan) [reply app/routers/event.py:30]: Probably not particularly. I just like the uniformity, but I guess it could be removed

Comment (OdeChan) [reply app/routers/event.py:30]: @yammesicka Probably not particularly. I just like the uniformity, but I guess it could be removed

File: app/routers/event.py [hunk 7]
```diff
@@ -25,16 +27,66 @@ async def eventedit(request: Request):
...
+async def eventview(request: Request, event_id: int,
+                    db: Session = Depends(get_db)):
+    try:
+        event = get_event_by_id(db, event_id)
+    except NoResultFound:
+        raise HTTPException(status_code=404, detail="Event not found")
+    except MultipleResultsFound:
```
Comment (efratush) [inline app/routers/event.py:37]: There can be no multiplication id in db

Comment (OdeChan) [reply app/routers/event.py:37]: True, I used the exception in case something went terribly wrong and somehow an event id did duplicate. When using .one() this will throw this exception. Using .first() would have return an arbitrary one, which would be more dangerous as it would potentially show the wrong event to the user

File: app/routers/event.py [hunk 8]
```diff
@@ -25,16 +27,66 @@ async def eventedit(request: Request):
...
+        raise HTTPException(status_code=500, detail="Multiple events found")
+
+    participants = get_participants_emails_by_event(db, event_id)
+
+    db.delete(event)  # Delete event
```
Comment (efratush) [inline app/routers/event.py:61]: Why did you remove Try?

File: app/routers/event.py [hunk 9]
```diff
@@ -25,16 +27,66 @@ async def eventedit(request: Request):
...
+    """Gets a single event by id"""
+    if not isinstance(db, Session):
+        raise AttributeError(
+            f'Could not connect to database. '
+            f'db instance type received: {type(db)}')
+    try:
+        event = db.query(Event).filter_by(id=event_id).one()
+    except NoResultFound:
+        raise NoResultFound(f"Event ID does not exist. ID: {event_id}")
+    except MultipleResultsFound:
```
Comment (efratush) [inline app/routers/event.py:84]: There can be no multiplication id in db

File: app/routers/event.py (hunk 7 above)
Comment (efratush) [reply app/routers/event.py:37]: This is not a possible situation on the part of db

File: app/routers/event.py (hunk 8 above)
Comment (OdeChan) [reply app/routers/event.py]: I moved it onto get_event_by_id, it will make sure that the event exists before trying to delete it. If I missed an exception, let me know :)

File: app/routers/event.py (hunk 9 above)
Comment (OdeChan) [reply app/routers/event.py:83]: Answered above

File: app/routers/event.py (hunk 7 above)
Comment (efratush) [reply app/routers/event.py:37]: Maybe because there is nothing general that will catch, it is right to leave the catch

File: app/routers/event.py (hunk 8 above)
Comment (OdeChan) [reply app/routers/event.py]: Ok, I think I fixed it no, sorry :)

**Review by yammesicka — APPROVED** [review]
